src/main/java/io/codelex/flightplanner/domain/Flight.java

```diff
@@ -1,54 +1,64 @@
 package io.codelex.flightplanner.domain;
 
 import com.fasterxml.jackson.annotation.JsonFormat;
-import jakarta.validation.Valid;
+import jakarta.persistence.*;
 import jakarta.validation.constraints.NotBlank;
 import jakarta.validation.constraints.NotNull;
-import org.springframework.format.annotation.DateTimeFormat;
-import org.springframework.lang.NonNull;
+import org.hibernate.annotations.Cascade;
 
-
-import java.text.DateFormat;
 import java.time.LocalDateTime;
-import java.time.format.DateTimeFormatter;
 import java.util.Objects;
-import java.util.Random;
 
+
+@Entity
+@Table(name = "flights")
 public class Flight {
-    private long id;
 
-    @NotBlank
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Integer id;
+
+    @NotNull
+    @ManyToOne
+    @Cascade(org.hibernate.annotations.CascadeType.ALL)
     private Airport from;
 
-    @NotBlank
+    @NotNull
+    @ManyToOne
+    @Cascade(org.hibernate.annotations.CascadeType.ALL)
     private Airport to;
 
+    @Column(name = "carrier", columnDefinition = "VARCHAR(255)")
     @NotBlank
     private String carrier;
 
+    @Column(name = "departure_time", columnDefinition = "VARCHAR(255)")
```

Comment:
Kolonnas tipam nebūtu jābūt VARCHAR (šeit vispār to column definition var ņemt ārā), tāpat arī redzu, ka DB tabulā Tev ir varchar - tas arī jāņem nost, tur jābūt normālam datu tipam.

Reply:
Paskatījos - tipu maiņa Tev pačakarēs nedaudz Tavu custom query:
`@Query("SELECT DISTINCT f FROM Flight f WHERE f.from.airport = :from AND f.to.airport = :to AND CAST(f.departureTime AS LocalDate) = :departureTime")`

Bet nekas - salabosi :)